### PR TITLE
Fix a bug when putting an event with metadata in event queue

### PR DESCRIPTION
**Pull request by Jyouping**
CoordinateEvents are getting deduped in the CoordinatorEventBlockingQueue based on the type. However, events with metadata shouldn't been deduped just based on the type. This change fixes the issue by only deduping the events with metadata. 

Going forward, we can consider dedup for eventMetadata as well. But this will requires eventMetadata object to implemented a proper hashcode logic.

Important: DO NOT REPORT SECURITY ISSUES DIRECTLY ON GITHUB.  
For reporting security issues and contributing security fixes,  
please, email security@linkedin.com instead, as described in  
the contribution guidelines.

Please, take a minute to review the contribution guidelines at:  
https://github.com/linkedin/Brooklin/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: datastream-server/src/main/java/com/linkedin/datastream/server/CoordinatorEventBlockingQueue.java [hunk 1]
```diff
@@ -46,8 +46,11 @@ public synchronized void put(CoordinatorEvent event) {
       }
     }
 
-    // always overwrite the event in the map
-    _eventMap.put(event.getType(), event);
+    // we put into the eventMap for dedup only if there is no metadata
+    if (event.getEventMetadata() == null) {
```
Comment (gauravrkulkarni) [inline datastream-server/src/main/java/com/linkedin/datastream/server/CoordinatorEventBlockingQueue.java]: We can go ahead with this for time being. In the long term we probably want the dedup logic to take metadata into consideration. With this implementation we have inconsistency that only events without metadata get deduped. Also there is no way for someone to know about this behavior unless checking out the code.

File: datastream-server/src/main/java/com/linkedin/datastream/server/CoordinatorEvent.java [hunk 2]
```diff
@@ -8,6 +8,9 @@
 
 /**
  * Represents different event types inside {@link Coordinator}.
+ *
+ * CoordinatorEvent will be deduped in the event queue based on the event type
```
Comment (gauravrkulkarni) [inline datastream-server/src/main/java/com/linkedin/datastream/server/CoordinatorEvent.java]: Could you please add description for what the bug was?

Comment (Jyouping) [reply datastream-server/src/main/java/com/linkedin/datastream/server/CoordinatorEvent.java]: added a link

File: datastream-server/src/main/java/com/linkedin/datastream/server/CoordinatorEventBlockingQueue.java (hunk 1 above)
Comment (Jyouping) [reply datastream-server/src/main/java/com/linkedin/datastream/server/CoordinatorEventBlockingQueue.java]: Deduping eventMetadata will requires assumptions to have the metadata properly implement the hashcode. This should be covered in the future implementation.

**Pull request merged**